pkg/crcontrollers/node/controller.go

```diff
@@ -453,6 +453,14 @@ func (bmc *Controller) updateNodeLabelsAndAnnotation(k8sNode *coreV1.Node, nodeU
 	if !bmc.externalAnnotation && ok {
 		if val == nodeUUID {
 			ll.Tracef("%s value for node %s is already %s", bmc.annotationKey, k8sNode.Name, nodeUUID)
+		}
+		// Use old UUID from the annotations
+		if val != nodeUUID {
```

Comment:
When will you updated nodes CRD with discovered UUID?

Reply:
I dont get. Could you explain?